### PR TITLE
Add detection for CVE-2024-47575

### PR DESCRIPTION
**Pull request by ybadaoui-ostorlab**
This PR introduces detection logic for **CVE-2024-47575**, a critical vulnerability in FortiManager caused by missing authentication for critical functions.

#### Detection Mechanism
The `check` function performs the following steps to determine if the target is vulnerable:
1. **Initial Requests**:
   - Sends `get ip` and `get auth` requests to establish the session.
2. **Vulnerable Case**:
   - If the response to the `get file_exchange` request includes a non-empty `remoteid` value, the target is identified as vulnerable, and a finding is reported.
3. **Secure Case**:
   - If the `remoteid` is missing or empty, no vulnerabilities are reported.

### CODE REVIEW

File: agent/exploits/cve_2024_47575.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
+from typing import List
```
Comment (PiranhaSa) [inline agent/exploits/cve_2024_47575.py:1]: can you add docstring please

Comment (nmasdoufi-ol) [inline agent/exploits/cve_2024_47575.py:1]: use `list` instead of `List`

File: tests/exploits/cve_2024_47575_test.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+from unittest.mock import patch, MagicMock
+from agent.exploits import cve_2024_47575
+from agent import definitions
+from typing import Callable
```
Comment (nmasdoufi-ol) [inline tests/exploits/cve_2024_47575_test.py:4]: No direct class call.

**Review by nmasdoufi-ol — COMMENTED** [review]
Good job pulling this detection together 👍

File: agent/exploits/cve_2024_47575.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+file_exch_cmd=put_json_cmd
+
+\0""".replace(b"\n", b"\r\n").replace(
+    b"REPLACE_LOCAL_ID", str(random.randint(100, 999)).encode()
+)
+
+
+def sendmsg(socket: ssl.SSLSocket, request: bytes) -> bytes:
+    """Send a message over an SSL socket and read the response."""
+    message = struct.pack(">II", 0x36E01100, len(request) + 8) + request
```
Comment (rabsondev) [inline agent/exploits/cve_2024_47575.py:80]: What is **`8`**?

File: agent/exploits/cve_2024_47575.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+chan_window_sz=32768
+deflate=gzip
+file_exch_cmd=put_json_cmd
+
+\0""".replace(b"\n", b"\r\n").replace(
+    b"REPLACE_LOCAL_ID", str(random.randint(100, 999)).encode()
+)
+
+
+def sendmsg(socket: ssl.SSLSocket, request: bytes) -> bytes:
```
Comment (rabsondev) [inline agent/exploits/cve_2024_47575.py:78]:
```suggestion
def send_msg(socket: ssl.SSLSocket, request: bytes) -> bytes:
```

File: tests/exploits/cve_2024_47575_test.py (hunk 2 above)
Comment (rabsondev) [inline tests/exploits/cve_2024_47575_test.py:4]: Please re-organise the imports.

Comment (deadly-panda) [reply tests/exploits/cve_2024_47575_test.py:4]: Typing and collections.abc are the exception: From google's python style sheet:
image

Comment (ybadaoui-ostorlab) [reply tests/exploits/cve_2024_47575_test.py:4]: sure!

File: agent/exploits/cve_2024_47575.py (hunk 3 above)
Comment (ybadaoui-ostorlab) [reply agent/exploits/cve_2024_47575.py:80]: The number of the header bytes

File: agent/exploits/cve_2024_47575.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        return b""
+
+    if len(hdr) != 8:
+        return hdr
+
+    try:
+        magic, size = struct.unpack(">II", hdr)
+        return socket.read(size)
+    except TimeoutError:
+        return b""
```
Comment (adnaneserrar) [inline agent/exploits/cve_2024_47575.py:97]: shouldn't we add an error logging here?

Comment (oussamaessaji) [reply agent/exploits/cve_2024_47575.py:97]: +1

**Review by 3asm — APPROVED** [review]

**Pull request merged**